### DISCUSSION TITLE
Suggest people use the JIT on PHP 8+

### DISCUSSION THREAD

**Discussion by iamacarpet** [PHP-CS-Fixer/PHP-CS-Fixer#7941] [Feature ideas]
## Feature request

It might be helpful for others to suggest that people enable the JIT when using this project with PHP 8+

We had a legacy project that was taking ~375s per run (without the cache file), but enabling the JIT brings this down to ~150s.

We are running with:

```
php -n -dmemory_limit=12G -dzend_extension=opcache.so -dopcache.enable_cli=On -dopcache.jit_buffer_size=128M vendor/bin/php-cs-fixer fix --config=...
```

**Comment by SpacePossum** [comment]
Hi and thanks for sharing!

Interesting numbers, when you say enabling JIT do you mean enabling the op cache on cli? Sorry for asking, I haven't looked into JIT configuration (yet) I'm afraid...

**Reply by Wirone** [reply]
@iamacarpet answered:

> Kind of, opcache & JIT are tied together in their config options, but perform different functions.
> 
> opcache is a cache of ‘code -> byte code’
>
> JIT is a cache of ‘byte code -> machine code’
>
> Basically bridges the performance gab between an interpreted language like PHP and a compiled language like C++ (although not entirely).

---

(Moved to thread to utilise discussions better)

**Comment by keradus** [comment]
Maybe it's sth we can use when spinning worker command ?

**Reply by Wirone** [reply]
I am not sure if it's related to parallel runner. It should rather be verified in main process and reported regardless of the runner used.

**Reply by keradus** [reply]
- main process can report it to end user to enable it
- main process can enable it themselve for worker processes, can't it ?

**Reply by Wirone** [reply]
Ah, I thought about reporting from workers, but you mean setting this in process' command to ensure it's enabled. Yeah, that could work, worth checking. But I won't do this in #7777, it's an improvement that can be implemented later ☺️.

**Reply by iamacarpet** [reply]
Thanks for posting a reference to #7777, just been catching up.

It looks like instead of enabling the required `php.ini` config options as command line arguments, when running in a container (like with a CI/CD type setup), it should be possible to just write out to the `php.ini` file before running the command.

It would be interesting to see in your benchmarks for #7777 what the difference is with JIT vs parallel, then JIT + parallel for maximum improvement.

**Reply by Wirone** [reply]
@iamacarpet I believe you can fiddle with it if you want, just checkout my branch and modify `ProcessFactory` accordingly. It would certainly be helpful 😊.
